src/NosCore.GameObject/Providers/GuriProvider/Handlers/EmoticonHandler.cs

```diff
@@ -38,7 +38,7 @@ public void Execute(RequestData<GuriPacket> requestData)
             if (requestData.Data.VisualEntityId.GetValueOrDefault() == requestData.ClientSession.Character.CharacterId)
             {
                 requestData.ClientSession.Character.MapInstance.Sessions.SendPacket(
-                    requestData.ClientSession.Character.GenerateEff(requestData.Data.Data +
+                    requestData.ClientSession.Character.GenerateEff((int)requestData.Data.Data +
```

Comment:
It needs to be converted

Reply:
No you need to check if not null before or it will crash on false packet

Reply:
How can that happen? I mean, up there in line 29 we have some "if" that will check that, no?

Reply:
Right but it make it difficult to follow.

Reply:
Mmm... Then what I need to change to make it more easy to follow?